eZ/Bundle/EzPublishCoreBundle/EventListener/ExceptionListener.php

```diff
@@ -29,10 +46,38 @@ public static function getSubscribedEvents()
     public function onKernelException(GetResponseForExceptionEvent $event)
     {
         $exception = $event->getException();
+
         if ($exception instanceof NotFoundException) {
-            $event->setException(new NotFoundHttpException($exception->getMessage(), $exception));
-        } elseif ($exception instanceof UnauthorizedException || $exception instanceof ForbiddenException) {
-            $event->setException(new AccessDeniedHttpException($exception->getMessage(), $exception));
+            $event->setException(new NotFoundHttpException($this->getTranslatedMessage($exception), $exception));
+        } elseif ($exception instanceof UnauthorizedException) {
+            $event->setException(new AccessDeniedHttpException($this->getTranslatedMessage($exception), $exception));
+        } elseif ($exception instanceof BadStateException || $exception instanceof InvalidArgumentException) {
+            $event->setException(new BadRequestHttpException($this->getTranslatedMessage($exception), $exception));
+        } elseif ($exception instanceof TranslatableExceptionInterface) {
```

Comment:
Should `instanceof TranslatableExceptionInterface` check be here since it is already in `getTranslatedMessage` method? Besides, this check means that execptions not implementing `TranslatableExceptionInterface` will never be converted to `HttpException` ?